src/coreclr/jit/optimizer.cpp

```diff
@@ -4577,36 +4591,23 @@ PhaseStatus Compiler::optUnrollLoops()
                 printf(", %d failures due to block cloning", unrollFailures);
             }
             printf("\n");
-            if (anyNestedLoopsUnrolled)
-            {
-                printf("At least one unrolled loop contains nested loops; recomputing loop table\n");
-            }
         }
 #endif // DEBUG
 
-        // If we unrolled any nested loops, we rebuild the loop table (including recomputing the
-        // return blocks list).
-        //
-        if (anyNestedLoopsUnrolled)
-        {
-            fgUpdateChangedFlowGraph(FlowGraphUpdates::COMPUTE_DOMS | FlowGraphUpdates::COMPUTE_RETURNS |
-                                     FlowGraphUpdates::COMPUTE_LOOPS);
-        }
-        else
-        {
-            fgUpdateChangedFlowGraph(FlowGraphUpdates::COMPUTE_DOMS);
-        }
-
-        m_dfsTree = fgComputeDfs();
+        // We left the old loop unreachable as part of unrolling, so get rid of
+        // those blocks now.
+        fgDfsBlocksAndRemove();
         optFindNewLoops();
 
+        fgDomsComputed = false;
+        fgRenumberBlocks();
```

Comment:
Not sure why this would be necessary, but there is a preexisting `fgDebugCheckBBlist` below that validates sequential `bbNum` values. Something for a future PR.